### ISSUE TITLE
Fasta output doesn't meet specification

### ISSUE THREAD

**Issue by SGSSGene** [seqan/seqan3#2767]
A fasta file looks roughly like this:
```
>SequenceId
ACGTACGT
```

When using seqan3 to output a fasta file, the file will have the content
```
> SequenceId
ACGTACGT
```
Note the extra space in between '>' and the sequence id.

From what I can see in the unit test this seems to be intentional behavior https://github.com/seqan/seqan3/blob/master/test/unit/io/sequence_file/sequence_file_format_fasta_test.cpp .

# Does this meet the fasta specification?
I am not sure. The german wiki page of the fasta description claims that there should be no space https://de.wikipedia.org/wiki/FASTA-Format
Contrary to the NCBI description, that doesn't make any statements about this https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp
Some tools like `razers3` only take the string until the first space as the header (Also a bug?), which in this case is empty.
This leads to results without identification strings.

**Comment by eseiler** [comment]
It's intended and configurable: https://docs.seqan.de/seqan/3-master-user/structseqan3_1_1sequence__file__output__options.html#ac292fdadacfa875bb187d30fc1996b4d

Not sure about whether space should be the default.

**Comment by SGSSGene** [comment]
Interesting. Yes this works, havn't seen it. What I would like to do is read a fasta file, adjust some sequences and write the fasta file again and make no changes to the sequence identifier. Reading the docs shows that reading a file will remove the blanks at the beginning? Is there a way to detect this?

**Comment by eseiler** [comment]
* There isn't really the *one* FASTA specification
* In general, an arbitrary identifier may follow the `>`
* No space after `>` seems way more common, I actually didn't find anything with `> `
* http://genetics.bwh.harvard.edu/pph/FASTA.html says that there *should* be no whitespace, but that's the only source I could find that specifically talks about a whitespace after `>`
* The [commit (Jul 9, 2018)](https://github.com/seqan/seqan3/commit/d0cba90d040249229d168e54206fe2b378597b96#diff-ade49c30dcbe6597145c0cadc82a7557b0a23940624098d4d5a2c107d8fc5741R53) that added the option set it to `true`
* I couldn't find anything that indicates that there was a discussion about having a whitespace or not

So, I think:
* No whitespace should be the default
* There should be a cookbook (or in tutorial?) about dis-/enabling the option
* **However**, it's also an almost unnoticeable API break

Also:
* We should fix RazerS3, I think I had the same problem once and found the same solution with the option 😅
* As far as I remember, RazerS3 shortens the identifier by splitting at the first whitespace - so there should be a check for a whitespace after `>`
---

>Is there a way to detect this?

What are you specifically referring to?
Probably not with the API...

**Comment by SGSSGene** [comment]
I changed the default for seqan3 here https://github.com/seqan/seqan3/pull/2769 .
I am not changing razerS3 because otherwise it is inconsistent with all the other seqan2 apps and changing/testing them all is to much.

**Comment by SGSSGene** [comment]
> > Is there a way to detect this?
> 
> What are you specifically referring to?
> Probably not with the API...

I added an option and changed the default: https://github.com/seqan/seqan3/pull/2770
This will allow an easy round trip of a fasta file without changing anything, when using the default values.

**Comment by marehr** [comment]
Thank you @eseiler for the write-up. I think the reason, why we chose whitespace after `>` is simply because seqan2 did it and no one reported a bug for that behaviour.

I think because of the convention to SPLIT the identifier and additional information AFTER the first whitespace, there might be implementations that get confused that `> identifier more|arguments|` contains a whitespace BEFORE the identifier. I think it is sound to change the default behaviour.

But, I think @SGSSGene's original wish that is to write a fasta file with the exact same "specification" as it was read in, is currently NOT solved. We just change the default behaviour, s.t. it matches now his use case.


If we read in a file

```
>hello some|identifier|in|id|and|wrap40|
CGCGAACGCCAGTACACGTTATCATGCTAAGTAGCGCTGC
AGCCTACAGAGATGACTGGAATGGGTGTGACACCCGAGGG
CGCCACCGATGCGAGCACGG
```
we currently write out something like this

```
> hello some|identifier|in|id|and|wrap40|
CGCGAACGCCAGTACACGTTATCATGCTAAGTAGCGCTGCAGCCTACAGAGATGACTGGAATGGGTGTGACACCCGAGGG
CGCCACCGATGCGAGCACGG
```
which has a wrap of 80 and a space after ID (or whatever the real defaults are)

What we actually want is to recognise the parameters a file was written out and give the user an API to write out a file with the same specification. This obviously has limits as fasta is more or less unspecified, and we accept also "invalid" fasta files.

The only kind of specification is enforced by fasta indices `.fai`, see http://www.htslib.org/doc/faidx.html.

> The samtools implementation uses the first word of the ">" header line text (i.e., up to the first whitespace character, having skipped any initial whitespace after the ">") as the NAME column. 

Even that format allows for each sequence a different wrapping and line-ending. I guess to allow `cat`ing different fasta files from different sources into one big fasta file.

So at least for fasta files this seems to be an option PER record and NOT per file (as it is right now) to produce the "same" output file from any given input file.

**Comment by SGSSGene** [comment]
This is fixed with #2869 and #2770
